code/modules/mob/living/carbon/xenomorph/strains/castes/warrior/shielder.dm

```diff
@@ -0,0 +1,602 @@
+		/datum/action/xeno_action/activable/plates_slam, //4th
+		/datum/action/xeno_action/onclick/reflective_shield, //5th
+	)
+
+	behavior_delegate_type = /datum/behavior_delegate/warrior_shielder
+
+/datum/xeno_strain/shielder/apply_strain(mob/living/carbon/xenomorph/warrior/warrior)
+	warrior.damage_modifier -= XENO_DAMAGE_MOD_SHIELDER
+	warrior.explosivearmor_modifier += XENO_EXPLOSIVE_ARMOR_TIER_1
+	warrior.add_plasma += XENO_PLASMA_TIER_2
+	warrior.speed += XENO_SPEED_TIER_1
+	warrior.tackle_min_modifier += 2
+	warrior.tackle_max_modifier += 1
+
+	warrior.recalculate_everything()
+
+//
+// Passive benefits
+//
+
+/datum/behavior_delegate/warrior_shielder
+	name = "Shielder Warrior Behavior Delegate"
+
+	var/frontal_armor = 10
+	var/side_armor = 5
+
+/datum/behavior_delegate/warrior_shielder/append_to_stat()
+	. = list()
+	. += "Front Armor: +[frontal_armor + bound_xeno.front_plates]"
+	. += "Side Armor: +[side_armor + bound_xeno.side_plates]"
+
+/datum/behavior_delegate/warrior_shielder/add_to_xeno()
+	RegisterSignal(bound_xeno, COMSIG_XENO_PRE_CALCULATE_ARMOURED_DAMAGE_PROJECTILE, PROC_REF(apply_directional_armor))
+
+/datum/behavior_delegate/warrior_shielder/proc/apply_directional_armor(mob/living/carbon/xenomorph/xeno, list/damagedata)
+	SIGNAL_HANDLER
+	var/projectile_direction = damagedata["direction"]
+	if(xeno.dir & REVERSE_DIR(projectile_direction))
+		damagedata["armor"] += frontal_armor
+	else
+		for(var/side_direction in get_perpen_dir(xeno.dir))
+			if(projectile_direction == side_direction)
+				damagedata["armor"] += side_armor
+				return
+
+/datum/behavior_delegate/warrior_shielder/on_update_icons()
+	if(bound_xeno.stat == DEAD)
+		return
+
+	if(bound_xeno.enclosed_plates && bound_xeno.health > 0)
+		bound_xeno.icon_state = "[bound_xeno.get_strain_icon()] Warrior Shield"
+		return TRUE
+
+//
+// 1st ability
+//
+
+/datum/action/xeno_action/verb/verb_toggle_plates()
+	set category = "Alien"
+	set name = "Toggle Encasing Plates"
+	set hidden = TRUE
+	var/action_name = "Toggle Plates Defense"
+	handle_xeno_macro(src, action_name)
+
+/datum/action/xeno_action/onclick/toggle_plates
+	name = "Toggle Encasing Plates"
+	action_icon_state = "crest_defense"
+	macro_path = /datum/action/xeno_action/verb/verb_toggle_plates
+	action_type = XENO_ACTION_ACTIVATE
+	xeno_cooldown = 3 SECONDS
+	ability_primacy = XENO_PRIMARY_ACTION_1
+
+	var/speed_debuff = 1
+
+/datum/action/xeno_action/onclick/toggle_plates/use_ability()
+	var/mob/living/carbon/xenomorph/xeno = owner
+	if(!istype(xeno))
+		return
+
+	if(!xeno.check_state())
+		return
+
+	if(!action_cooldown_check())
+		return
+
+	xeno.enclosed_plates = !xeno.enclosed_plates
+
+	if(xeno.enclosed_plates)
+		to_chat(xeno, SPAN_XENOWARNING("We raise our plates and form shield."))
+		RegisterSignal(owner, COMSIG_XENO_PRE_CALCULATE_ARMOURED_DAMAGE_PROJECTILE, PROC_REF(check_directional_armor))
+		xeno.ability_speed_modifier += speed_debuff
+		xeno.mob_size = MOB_SIZE_BIG //knockback immune
```

Comment:
This gives it more then knockback immunity, if your only reason for giving it SIZE_BIG is knockback immunity theres other ways to do that